src/tasks/ivfm.py

```diff
@@ -0,0 +1,134 @@
+import os
+from functools import partial
+from pathlib import Path
+
+import numpy as np
+import SimpleITK as sitk
+import torch
+from einops.einops import rearrange
+from mat73 import loadmat
+
+from src.models.components.mlp import MLP
+from src.models.pinns_ivfm import PINNs_ivfm
+
+# Set device for training
+DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+# Seed for reproducibility
+np.random.seed(12345)
+torch.manual_seed(12345)
+if torch.cuda.is_available():
+    torch.cuda.manual_seed(12345)
+
+# Create data
+# Set current working directory
+cwd = Path(__file__).resolve().parent
+os.chdir(cwd)
+patient = "../../data/patient_06_1.mat"
+data = loadmat(patient)
+fname = os.path.basename(patient)[:-4]
+spacing = [0.5, 0.5, 1]
```

Comment:
Are we sure that the spacing is the same for each data / patient ? Wouldn't it be better to put this information in the ".mat" file and load it here ?

Reply:
Spacing is only used for saving the nifti file and for illustration purpose when the file is opened in ITK snap. Since we did not interpolate the image, the spacing is actually “useless” and won't affect the results.

Reply:
This is true if all the native image are isotropic, is it the case ?